python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPrivate.qll

```diff
@@ -3,6 +3,16 @@ private import DataFlowPublic
 import semmle.python.SpecialMethods
 private import semmle.python.essa.SsaCompute
 private import semmle.python.dataflow.new.internal.ImportStar
+// Since we allow extra data-flow steps from modeled frameworks, we import these
+// up-front, to ensure these are included. This provides a more seamless experience from
+// a user point of view, since they don't need to know they need to import a specific
+// set of .qll files to get the same data-flow steps as they are used to seeing. This
+// also ensures that we don't end up re-evaluating data-flow because it has different
+// global steps in some configurations.
+//
+// This matches behavior in C#.
+private import semmle.python.Frameworks
+private import semmle.python.Concepts
```

Comment:
If we import `Concepts`, do we also need to import `Frameworks`?

Reply:
I am a bit wary of this, but only because I vaguely remember a debug sessions with @tausbn where we needed to avoid it. Perhaps he can recall some details..

Reply:
What you mention _is_ stirring some disused neurons at the back of my brain, but unfortunately I can't remember off-hand what the issue was.

I think the use here is probably fine. I guess the only objection would be that you can't turn them off again. No matter what, you'll get the ORM-based data-flow steps, even if you don't want them. 

(I guess this _might_ matter if someone were to, say, define a local version of `django` that doesn't use the same way of passing data through models. I think this is very unlikely, though.)

Reply:
> If we import Concepts, do we also need to import Frameworks?

Technically no, since we know that Concepts imports Frameworks... if you would like me to change that part, I can certainly do that :+1:

> I think the use here is probably fine. I guess the only objection would be that you can't turn them off again. No matter what, you'll get the ORM-based data-flow steps, even if you don't want them.

Yes, agreed that there _are_ downsides to this. I think this won't be a problem for 99.999999% of the cases, so I would much rather have the positives effects mentioned in the comment from this :blush:

Reply:
Alright, let's ship to learn on that one. (And I hope I will not regret this when doing flow summaries...).

For the import, I think I would prefer to be minimal, since seeing both confused me. I do see the situation where we no longer need concepts and forget to import frameworks...but that should be caught quickly...

Reply:
```suggestion
private import semmle.python.Frameworks
```